### PR TITLE
[cookies] Add `--cookies-from-browser` support for Firefox from Microsoft Store

### PR DESCRIPTION
**Pull request by wesson09**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Provide support for cookies from Firefox Browser when installed from Microsoft Store

Fixes #

<details open><summary>Firefox browser if installed first from Microsoft Store under Windows 11 will store user data into 
another folder. Add this folder to the search list. </summary> 

[Firefox Documentation](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data)

### Before submitting a *pull request* make sure you have:
- [ x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/version.py [hunk 1]
```diff
@@ -12,4 +12,4 @@
 
 ORIGIN = 'yt-dlp/yt-dlp'
 
-_pkg_version = '2024.12.03'
+_pkg_version = '2024.12.04'
```
Comment (seproDev) [inline yt_dlp/version.py:15]:
```suggestion
_pkg_version = '2024.12.03'
```

File: yt_dlp/cookies.py [hunk 2]
```diff
@@ -195,7 +195,10 @@ def _extract_firefox_cookies(profile, container, logger):
 
 def _firefox_browser_dirs():
     if sys.platform in ('cygwin', 'win32'):
-        yield os.path.expandvars(R'%APPDATA%\Mozilla\Firefox\Profiles')
+        yield from map(os.path.expandvars, (
+            R'%APPDATA%\Mozilla\Firefox\Profiles',
+            R'%LOCALAPPDATA%\Packages\Mozilla.Firefox*\LocalCache\Roaming\Mozilla\Firefox\Profiles',
```
[the remark this answers is not in the document]
Comment (wesson09) [reply yt_dlp/cookies.py:200]: [Official firefox documentation](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data) refer only to the start of the path "%LOCALAPPDATA%\Packages\Mozilla.Firefox" so I have assumed the ending part may change across installations.

[the remark this answers is not in the document]
Comment (seproDev) [reply yt_dlp/cookies.py:200]: The package family name is handed out by Microsoft. I would consider it constant.

[the remark this answers is not in the document]
Comment (wesson09) [reply yt_dlp/cookies.py:200]: So be it.

**Review by seproDev — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

**Comment by candrapersada** [comment]
`--cookies-from-browser` support for Firefox from `winget`?

**Review by Zulu-1540 — COMMENTED** [review]
There problem download the apps, seems really harmful

**Comment by candrapersada** [comment]
why `--cookies-from-browser` not working in https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Mozilla/Firefox/MSIX/147.0.2/Mozilla.Firefox.MSIX.installer.yaml

`%LOCALAPPDATA%\Packages\Mozilla.MozillaFirefox_jag0gd4e3s9p2\LocalCache\Roaming\Mozilla\Firefox\Profiles`

**Pull request merged**